datafusion/functions/src/string/common.rs

```diff
@@ -251,6 +252,22 @@ impl<'a> ColumnarValueRef<'a> {
     }
 }
 
+pub trait StringArrayType<'a>: ArrayAccessor<Item = &'a str> + Sized {
```

Comment:
This makes sense -- thank you @tlm365  and @Omega359  -- I'll make a PR to add some additional comments to this trait as well

Reply:
https://github.com/apache/datafusion/pull/12027 is a PR to document this trait and how to use it